experimental/PropertyDDS/packages/property-common/src/error_objects/operationError.ts

```diff
@@ -2,106 +2,89 @@
  * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
  * Licensed under the MIT License.
  */
-/* eslint-disable no-use-before-define */
+
 /**
  * An operation error maintains additional information compared to a plain {@link #Error}:
  * - The operation name
  * - A status code
  * - Extensible flags. {@see ExtendedError.FLAGS}.
  */
+import _ from "lodash";
+import { FlaggedError } from "./flaggedError";
 
-var _ = require('lodash');
-var FlaggedError = require('./flagged_error');
+export class OperationError extends Error {
+    static FLAGS = FlaggedError.FLAGS;
+    public stack: string | undefined;
+    public readonly name: string;
 
-/**
- * Instantiates an OperationError, which mimics the {@link #Error} class with added properties
- * meant for reporting the result of operations.
- * @param {string} message The error message.
- * @param {?string} operation The operation name.
- * @param {?number} statusCode The operation result as a numerical status code.
- * @param {?number} flags Flags that characterize the error. See {@link FlaggedError#FLAGS}.
- * @constructor
- * @alias property-common.OperationError
- */
-var OperationError = function (message, operation, statusCode, flags) {
-    Error.call(this, message);
-    this.name = 'OperationError';
-    this.operation = operation;
-    this.statusCode = statusCode;
-    this.flags = flags || 0;
-    var stack = Error(message).stack;
+    /**
+       * Instantiates an OperationError, which mimics the Error class with added properties
+       * meant for reporting the result of operations.
+       * @param message - The error message.
+       * @param operation - The operation name.
+       * @param statusCode - The operation result as a numerical status code.
+       * @param flags - Flags that characterize the error. See {@link FlaggedError.FLAGS}.
+       */
+    constructor(
+        message?: string,
+        public operation?: string,
+        public statusCode?: number,
+        public flags: number = 0,
+    ) {
+        super(message);
+        Object.setPrototypeOf(this, OperationError.prototype);
```

Comment:
That is the working inheritance from the Error class.
https://stackoverflow.com/questions/41102060/typescript-extending-error-class
Failing existing tests otherwise.